src/components/tagList/AposTagListItem.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+  <li class="apos-tag-list__item">
+    <button :class="{'is-active' : active}" class="apos-tag-list__button" @click="click(tag)">
+      <transition name="slide-fade" mode="out-in" duration="100">
+        <Close v-if="this.active" class="apos-tag-list__icon apos-tag-list__icon--remove" :size="13"/>
+        <Tag v-else class="apos-tag-list__icon apos-tag-list__icon--tag" :size="13" />
+      </transition>
+      <span class="apos-tag-list__label">{{ tag.label }}</span>
+    </button>
+  </li>
+</template>
+
+<script>
+import AposTagListItem from './AposTagListItem.vue';
+import Tag from 'vue-material-design-icons/Label.vue';
+import Close from 'vue-material-design-icons/Close.vue';
+export default {
+  props: {
+    tag: Object
```

Comment:
I'm thinking we should start requiring more props where possible. This seems like a candidate for that.

Reply:
required this but i think in this case we assume it will always be fine because we're iterating over an array. if we dont get the right thing we'll get a type error.

I also required the array of tags be passed into the AposTagList, so with those two checks we'll either get a Required error or typeof error somewhere along the chain